dtls.c

```diff
@@ -2848,22 +2883,47 @@ dtls_send_client_hello(dtls_context_t *ctx, dtls_peer_t *peer,
     p += cookie_length;
   }
 
+  for (index = 0; handshake->cipher_suites[index] != TLS_NULL_WITH_NULL_NULL; ++index) {
+    dtls_cipher_t code = handshake->cipher_suites[index];
+    if (known_and_supported_cipher_suite(ctx, code, 1)) {
+        cipher_suites_size += sizeof(uint16);
+#ifdef DTLS_ECC
+        ecdsa = ecdsa || is_tls_ecdhe_ecdsa_with_aes_128_ccm_x(code);
+#endif /* DTLS_ECC */
+    }
+    // ignore not supported cipher-suite
+    // credentials callback is missing
+  }
+
+  if (cipher_suites_size == 0) {
+    dtls_crit("no supported cipher-suite provided!\n");
+    return dtls_alert_fatal_create(DTLS_ALERT_HANDSHAKE_FAILURE);
+  }
+
   /* add known cipher(s) */
-  dtls_int_to_uint16(p, cipher_size - 2);
+  dtls_int_to_uint16(p, cipher_suites_size);
   p += sizeof(uint16);
 
-  if (ecdsa) {
-    dtls_int_to_uint16(p, TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8);
-    p += sizeof(uint16);
-    dtls_int_to_uint16(p, TLS_ECDHE_ECDSA_WITH_AES_128_CCM);
-    p += sizeof(uint16);
+  for (index = 0; handshake->cipher_suites[index] != TLS_NULL_WITH_NULL_NULL; ++index) {
+    dtls_cipher_t code = handshake->cipher_suites[index];
+    if (known_and_supported_cipher_suite(ctx, code, 1)) {
+        dtls_int_to_uint16(p, code);
```

Comment:
Please use always 2-char indentation of code blocks.

Reply:
changed